src/OrleansRuntime/MembershipService/MembershipTableFactory.cs

```diff
@@ -22,50 +23,74 @@ public MembershipTableFactory(IServiceProvider serviceProvider, LoggerWrapper<Me
             this.logger = logger;
         }
 
+        private async Task<IMembershipTable> GetMembershipTableLegacy(GlobalConfiguration globalConfiguration)
+        {
+            IMembershipTable result = null;
+            switch (globalConfiguration.LivenessType)
+            {
+                case LivenessProviderType.MembershipTableGrain:
+                    result = await this.GetMembershipTableGrain();
+                    break;
+                case LivenessProviderType.SqlServer:
+                    result = AssemblyLoader.LoadAndCreateInstance<IMembershipTable>(Constants.ORLEANS_SQL_UTILS_DLL, this.logger, this.serviceProvider);
```

Comment:
Will this approach work? you are not configuring the strongly typed options for the particular providers... that's why I was suggesting having a configurator instead that gets loaded by the fact of using ClusterConfiguration in the builder

Reply:
You are absolutely right.. my bad. But I wonder how those LivenessTests in AzureUtils pass. Those tests are using the legacy way to create membershiptable. Maybe I need to add more tests.

Reply:
Ah. I think Jenkins wouldn't run those tests....